packages/shared/src/util/features.ts

```diff
@@ -324,6 +328,21 @@ export function mergeResultHasChanges(mergeResult: AutoMergeResult): boolean {
 
   return false;
 }
+export function listChangedEnvironments(
+  base: RulesAndValues,
+  revision: RulesAndValues
+) {
+  const environmentsList: string[] = [];
+  Object.keys(revision.rules)?.forEach((env) => {
```

Comment:
Environments can be added and removed from an org, but it won't update revisions.  This function should take in a list of valid environments from the org instead and use that to loop through.

Reply:
should we update the revisions so that we are not holding around this data or just mark it as deleted

Reply:
No need to update anything.  Doesn't matter if the revision has changes to an environment that doesn't exist anymore.  All we need to change here is to loop over the org's environments instead of assuming `Object.keys(revision.rules)` is up-to-date.